openedx/tests/xblock_integration/test_external_xblocks.py

```diff
@@ -0,0 +1,52 @@
+"""
+This will run tests on all XBlocks in the `xblock.test.v0`
+entrypoint. Did you notice something about that entry point? It ends
+with a v0. That means this is not finished. At some point, we might
+stop running v0 tests, replacing them with test case failures, and
+run v1 tests only.
+
+That be the dragon here.
+"""
+import pkg_resources
+
+
+class DuplicateXBlockTest(Exception):
+    '''
+    This exception is shown if there are multiple entry points with the same
+    class name for a test. In most cases, this means you have two versions
+    of the same XBlock installed, or two XBlocks with namespace collisions. In
+    either case, it'd be nice to resolve.
+    '''
+    pass
+
+
+class InvalidTestName(Exception):
+    '''
+    This means you have an entry point for a test that does not correspond
+    to a properly named test class. For example, if you cut-and-paste entry
+    points in `setup.py`, and forgot to repoint the class (so it points to
+    `DoneXBlock` instead of `TestDone`), or otherwise made an error, you
+    will see this exception.
+    '''
+    pass
+
+xblock_loaded = False  # pylint: disable=invalid-name
+
+for entrypoint in pkg_resources.iter_entry_points(group="xblock.test.v0"): # pylint: disable=no-member
+    plugin = entrypoint.load()
+    classname = plugin.__name__
+    if classname in globals():
+        raise DuplicateXBlockTest(classname)
+    if not classname.startswith("Test"):
+        raise InvalidTestName("Test class should start with 'Test': " + classname)
+    if not classname.replace("_", "").isalnum():
+        raise InvalidTestName("Python variables should be letters, numbers, and underscores: " + classname)
```

Comment:
I generally prefer to be a little overly conservative, but if you insist, I'll remove the extra check. I think it would minimize the impact of possible malicious tricks, but in this case, there is no immediate increase in security exposure, so I can kill it. It's just two lines of code, though. Your call.

I share the concern about blocks clobbering each others' tests, but I figure we can address that when we get to it. Right now, I have exactly two test cases (`RateXBlock` and `DoneXBlock`). It wouldn't be hard to rename the tests as they come in and similar, but I would be a little concerned about failing to catch corner cases:

* Introspection/dynamic code which refers to things by name.
* Absolute imports with overlapping names
* Same test class name vs. same test+package name vs. etc.
* Overlaps in identifiers in OLX
* Etc. 

There are a lot of these corner cases here, but without generating them, and making a lot of test cases, I'd have little confidence I'd address them correctly, and even so, I wouldn't be sure I thought of all them. For now, we have 50 XBlocks in the community, which is a little bit unwieldy for tests, but issues like this can be managed on a case-by-case basis.

Note that a lot of these already come up in XBlocks. If I have two blocks which both call themselves `<video>`, the OLX parsing becomes undefined.